### ISSUE TITLE
Wall-top Profiled Sweeps are not exported

### ISSUE THREAD

**Issue by DinaMavliutova** [Autodesk/revit-ifc#262]
Sweeps added to top of the wall via: Edit Type > Construction > Edit > Modify Vertical Structure > Sweeps
do not come through IFC export.
Revit:
![wall composites 1](https://user-images.githubusercontent.com/70137266/102329148-b20d7900-3f90-11eb-9c1c-5a77c1eb281f.jpg)
Solibri:
![image](https://user-images.githubusercontent.com/70137266/102329212-cc475700-3f90-11eb-9772-6ccb5af90554.png)

**Comment by AngelVelezSosa** [comment]
I don't think this is a generic issue - can you include a Revit file that shows your problem?

**Comment by DinaMavliutova** [comment]
![image](https://user-images.githubusercontent.com/70137266/103510210-6ce0b680-4e6d-11eb-934e-82d2898ff2a9.png)
another image describing where the sweep comes from, and the files, wall sample and sweep profile family
[Composite walls with top sweep.zip](https://github.com/Autodesk/revit-ifc/files/5763470/Composite.walls.with.top.sweep.zip)


> 
> 
> I don't think this is a generic issue - can you include a Revit file that shows your problem?

**Comment by tomekduda-tdbim** [comment]
Hi,

Searching for quite similar problem I've found maybe not solution but rather explanation (with some walkaround ideas) of the issue.

Look here:
https://knowledge.autodesk.com/support/revit-products/troubleshooting/caas/sfdcarticles/sfdcarticles/Autodesk-Revit-Sweeps-and-ifc-classes.html

And more importantly here:
https://www.revitforum.org/showthread.php/40156-Wall-sweeps-IFC-export

Nowadays I'm during project without specific BIM requirements so I'm not sure if I'll have enough time to solve the problem soon but when it came the right time for that...at first I'll try create that objects not using Revit Wall Sweeps (within wall definition or by creating "standalone" sweeps) - I'd rather start solving the problem by creating linebased family (or if it doesn't work as it is suggested inside second link - facebased but with ability to quite dynamic/flexible stretch length of the family) with solids assigned to subcategories (if you'd like do export each layer/solid separately to different IFC categories - than you could set proper mapping into IFC Export Options as you wish).

**Comment by dvrvb** [comment]
Uncheck "section box" while doing the IFC export, and the sweeps on top of the walls should be exported fine.
![image](https://user-images.githubusercontent.com/41319048/103875951-563b9900-50d3-11eb-8d34-3880141c7828.png)

**Comment by DinaMavliutova** [comment]
> Uncheck "section box" while doing the IFC export, and the sweeps on top of the walls should be exported fine.
> ![image](https://user-images.githubusercontent.com/41319048/103875951-563b9900-50d3-11eb-8d34-3880141c7828.png)
I need section box, as i want to show internal structure of the wall (layers). But I see your point.
Maybe the issue is that "wall-top sweeps are not exported when section box is in use".

**Comment by JOuellette-Autodesk** [comment]
@DinaMavliutova I'm following up to see that this has been resolved. Can you provide a file or test it again yourself?

**Comment by ArtemHnatko** [comment]
The issue is not reproducible in Revit 2023-2025. Wall-top profiled sweeps are successfully exported to IFC.

**Testing environment:**
- Autodesk Revit 2023 (v.23.1.60.0)
- Autodesk Revit 2024
- Autodesk Revit 2025

![Image](https://github.com/user-attachments/assets/8f071e53-dd4a-427e-b775-88c0d4742400)
